packages/gitmoji-changelog-core/src/index.spec.js

```diff
@@ -113,6 +124,20 @@ describe('changelog', () => {
     ])
   })
 
+  it('should filter some commits out', async () => {
+    gitRawCommits.mockReset()
+    mockGroup([uselessCommit, lipstickCommit])
+
+    gitSemverTags.mockImplementation(cb => cb(null, []))
+
+    const { changes } = await generateChangelog({ mode: 'init' })
+
+    expect(changes).toHaveLength(1)
+    expect(changes[0].groups).toHaveLength(1)
+    expect(changes[0].groups[0].commits).toHaveLength(1)
+    expect(changes[0].groups[0].commits[0]).toEqual(lipstickCommit)
```

Comment:
The last expect need the previous ones to be true so you can delete them, no ?

Reply:
Yeah maybe.
When I wrote those I was more about: "if jest tells me lines where expect is false, I could tell where there is a problem in the code"

And I thank it was better than a JS error.

What do you think?

Reply:
I guess i would write 4 different test for this purpose. Ping @Taranys

Reply:
Jest produces codeframes so you get a diff between what's expected and what you got.
Then you'll know exactly what is wrong in `changes`.

Reply:
```js
expect(changes).toEqual([{
  groups: [{ commits: [lipstickCommit] }],
}])
```

You could also use `expect.objectContaining` and `expect.arrayContaining` if you wish to ignore some parts of `changes`.

Reply:
Ok I'll try that, thank you @charlyx